scipy/special/xsf/tools.h

```diff
@@ -261,10 +258,271 @@ namespace detail {
      * See ContinuedFractionSeriesGenerator for details.
      */
     template <typename Generator, typename T = pair_value_t<generator_result_t<Generator>>>
-    XSF_HOST_DEVICE ContinuedFractionSeriesGenerator<Generator, T>
-    continued_fraction_series(Generator &cf) {
+    XSF_HOST_DEVICE ContinuedFractionSeriesGenerator<Generator, T> continued_fraction_series(Generator &cf) {
         return ContinuedFractionSeriesGenerator<Generator, T>(cf);
     }
 
+    /* Find initial bracket for a bracketing scalar root finder. A valid bracket is a pair of points a < b for
+     * which the signs of f(a) and f(b) differ. Assumes function is monotonic and it is known
+     * whether the function is increasing or decreasing. Passing in whether the function is
+     * increasing or not allows for simplication of logic and avoidance of some branches.
```

Comment:
It seems the code doesn’t rely on the monotonicity of $f$ to function correctly. It seems sufficient for the sign of $f$ to be monotone. Since there are probably interesting use cases where the function is not monotone but its sign is, is it worth updating the comment to clarify?

Reply:
This is a very good point. I was parroting something I read in the cdflib docs, but the bracket finder I wrote is different and doesn’t have the same requirements.  I’ll update the comment accordingly.